### ISSUE TITLE
Sigma - firmware updating / connection problems

### ISSUE THREAD

**Issue by S3D-r** [BCN3D/BCN3D-Cura-Windows#17]
I was not able to connect Sigma with "Cura for BNC3D":
1. Connecting starts...
   ![1 cura firmware update connecting](https://cloud.githubusercontent.com/assets/19798034/15856090/3b70cc7a-2cbc-11e6-9ab6-e40055d5dbba.jpg)
2. It seems that there is a connection!
   ![2 cura firmware update connected](https://cloud.githubusercontent.com/assets/19798034/15856116/5576075c-2cbc-11e6-8263-aa1100a993a2.jpg)
3. But no, when I try to update the Firmware I get this:
   ![3 cura firmware update connection error](https://cloud.githubusercontent.com/assets/19798034/15856129/606eaad8-2cbc-11e6-88fa-5b2e787e7e26.jpg)
4. The error log is:
   ![4 cura firmware update connection error log](https://cloud.githubusercontent.com/assets/19798034/15856142/6d73fae4-2cbc-11e6-9b38-4e8cf602c8e7.jpg)

**Comment by remelbourne** [comment]
@S3D-r This should have been fixed in the new version. Let me know if you still have the same problem!

**Comment by S3D-r** [comment]
Well, no matter how good "the new version" is, as I'm not able to upload the new version. Actually, that was my primary problem (and still is...).

**Comment by sigmatester** [comment]
I have the same problem. Is there any other way to upload firmware?

**Comment by S3D-r** [comment]
BCN3D is currently working on this problem (working hard, I hope...), I hope they will find a remedy soon. I'll let you know.

**Comment by remelbourne** [comment]
@S3D-r I meant the new Cura-BCN3D version. We redesigned the firmware updater and it should be working fine. For now it is only a beta version so please let us know if anything is still failing to work. You can find it on the following link: https://github.com/BCN3D/BCN3D-Cura-Windows/releases

@sigmatester You can upload firmware manually by first downloading the source files from this link https://github.com/BCN3D/BCN3DSigma-Firmware/releases and later in Cura-BCN3D go to "Machine -> Install custom firmware" and look for a ".hex" file in the source files you just downloaded which should be under the folder "Compiled Firmware". I recommend you download the latest beta version of Cura-BCN3D since the manual firmware installer did not work too well in previous versions.

Hope this helps! Do not hesitate to post any further questions.

**Comment by S3D-r** [comment]
Hi, remelbourne!  What does the bug-label mean? Does it mean that you propose a remedy to a problem that is based on some bug in s/w?  

I will give the beta-Cura a change... I'll report, how it went...

**Comment by remelbourne** [comment]
Hi @S3D-r, the "bug" label is just a way for us to keep track of issues. What happened to you in the first place was a bug in the code and so this helps us and users see if it already happened or it is something new.

**Comment by S3D-r** [comment]
I finally managed to update the firmware.  I had had problems with manual update (of f/w), the updating screen appeared but never progressed:

![firmare upload doesnt progress](https://cloud.githubusercontent.com/assets/19798034/18224950/cb380468-71ef-11e6-8af6-1e25c6e13e36.jpg)

The F/W file was not correct. The file v01-1.1.6.hex that I downloaded on 26th July, was 2349kBytes. Yesterday I checked Github and that file was 446kBytes. With that smaller hex-file I managed with f/w updating and the Sigma seems to work now. What a big relief!
